phonetest.py

```diff
@@ -200,18 +201,93 @@ def lookup(cls, phoneid, config_file, chunk):
+                branch = 'integration/' + branch
+            s = requests.Session()
+            url = 'https://hg.mozilla.org/' + branch + '/json-pushes?changeset=' + revision
+            r = s.get(url,
+                      headers={'Accept': 'application/json',
+                               'User-Agent': 'autophone'})
+
+            if r.status_code != 200:
+                logger.debug('Could not find pushlog for revision %s at %s' %
+                             (revision, url))
+                return set()
+
+            pushlog = r.json()
+            for pushid in pushlog:
+                try:
+                    changesets = pushlog[pushid]['changesets']
+                except:
+                    # We normally see a 'user' in this json blob, more might
+                    # be added in the future.
+                    continue
+                for changeset in changesets:
+                    #TODO: When Bug 1286353 is fixed, use json-rev for this,
+                    #      as this requires retrieving and scanning the whole
+                    #      diff.
+                    url = 'https://hg.mozilla.org/' + branch + '/raw-rev/' + changeset
+                    r = s.get(url,
+                              headers={'User-Agent': 'autophone'})
+                    if r.status_code == 200:
+                        for line in r.text.split('\n'):
+                            if line.find('/dev/null') != -1:
+                                continue
+                            if line.startswith('+++') or line.startswith('---'):
+                                # skip markers, space and leading slash
+                                changeset_files.add(line[6:])
+                    else:
+                        logger.debug('Could not find diff for revision %s at %s' %
+                                     (changeset, url))
+                        # We return an empty set here to force the test to be
+                        # run, in case the missing diff here contained files
+                        # we care about.
+                        return set()
+
+            return changeset_files
+
+        if has_run_if_changed and branch and revision:
+            changeset_files = get_changeset_files(branch, revision)
+
+        for test in tests:
+            # If changeset_files is empty, we will run the tests anyway.
+            # This is safer in terms of catching regressions and extra tests
+            # being run are more likely to be noticed and fixed than tests
+            # not being run that should have been.
+            if test.run_if_changed and changeset_files:
+                matched = False
+                for f in changeset_files:
+                    for d in test.run_if_changed:
+                        if f.startswith(d):
+                            matched = True
+                            break
+                if not matched:
+                    continue
+
```

Comment:
I think it's best to support being able to do a prefix match, so we can just write '/blah' in the manifest file and have it match every path under '/blah' rather than listing them out individually.

Set intersection would have been really clean though, I was part way through implementing it before I realized prefixes wouldn't work nicely.